test/built-ins/Temporal/Calendar/prototype/inLeapYear/string-throw-range-error.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.calendar.prototype.inleapyear
+description: Temporal.Calendar.prototype.inLeapYear throw RangeError while ISO8601 string is not a date.
+  and return true or false that year is leap year.
+  and return Array of the same content.
+info: |
+  1. Let calendar be the this value.
+  2. Perform ? RequireInternalSlot(calendar, [[InitializedTemporalCalendar]]).
+  3. Assert: calendar.[[Identifier]] is "iso8601".
+  4. If Type(temporalDateLike) is not Object or temporalDateLike does not have an [[InitializedTemporalDate]] or [[InitializedTemporalYearMonth]] internal slot, then
+  a. Set temporalDateLike to ? ToTemporalDate(temporalDateLike).
+  5. Return ! IsISOLeapYear(temporalDateLike.[[ISOYear]]).
```

Comment:
This test could add the text that throws the RangeError:

```
ParseTemporalDateString ( isoString )

1. Assert: Type(isoString) is String.
2. If isoString does not satisfy the syntax of a TemporalDateString (see 13.33), then
  a. Throw a RangeError exception.
3. Let result be ? ParseISODateTime(isoString).
4. Return the new Record { [[Year]]: result.[[Year]], [[Month]]: result.[[Month]], [[Day]]: result.[[Day]], [[Calendar]]: result.[[Calendar]] }.
```

And also the relations to the [iso8601](https://tc39.es/proposal-temporal/#sec-temporal-iso8601grammar) could be valuable here

Reply:
This can be shortened by saying there is no syntax produced in the [iso8601 grammar section](https://tc39.es/proposal-temporal/#sec-temporal-iso8601grammar) for these format variations that otherwise look like regular dates.